core/conn/jdbc_type2/native/SqlInterface.cpp

```diff
@@ -246,10 +246,14 @@ SQLRETURN GetJDBCValues( SQLItemDesc_def *SQLItemDesc,
 		SQLItemDesc->signType      = TRUE;
 		break;
 	case SQLTYPECODE_DATETIME:
+		// Calculate the length based on YYYY/MM/DD HH:MM:SS.ffffff
+        SQLItemDesc->maxLen = 20 + 6;
```

Comment:
The issue is not about getMemoryAllocInfo here. Since the maxLen is hard code to 26, so the length of the memory is always enough. 

When use MT-DCS, the maxLen will be passed to driver side. ODBC driver will check if the maxLen is equal to the real length of the value, for example , the maxLen of DATE should be 10.